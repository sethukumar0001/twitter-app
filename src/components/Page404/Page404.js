import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Typography, Button, withStyles } from '@material-ui/core';

const styles = theme => ({
  button: {
    marginTop: theme.spacing.unit * 2,
  },
});

class Page404 extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
  };

  render() {
    const { classes } = this.props;

    return (
      <React.Fragment>
        <Typography variant="display4">404</Typography>
        <Typography variant="headline">
          We could not find a page you are looking for :(
        </Typography>
        <Button
          variant="outlined"
          className={classes.button}
          component={Link}
          to="/"
        >
          Back Home
        </Button>
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(Page404);
