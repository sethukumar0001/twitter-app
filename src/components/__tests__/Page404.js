import React from 'react';
import renderer from 'react-test-renderer';
import Page404 from '../Page404';

jest.mock('react-router-dom');

describe('Page404', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Page404 />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
