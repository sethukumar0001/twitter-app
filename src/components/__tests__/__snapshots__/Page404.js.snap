// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 renders correctly 1`] = `
Array [
  <h1
    className="MuiTypography-root-2 MuiTypography-display4-3"
>
    404
</h1>,
  <h1
    className="MuiTypography-root-2 MuiTypography-headline-7"
>
    We could not find a page you are looking for :(
</h1>,
]
`;
